modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx

```diff
@@ -536,20 +548,67 @@ const ClaimsTable = () => {
 	const title = `Claims (${totalCount})`;
 
 	// eslint-disable-next-line react-hooks/exhaustive-deps
-	const updateCheckedProduct = (currentFilterName: string) => {
-		const isChecked = checkedStateProduct.map(
+	const genericUpdateCheckedStatus = (
+		currentFilterName: string,
+		checkedArray: boolean[],
+		items: ItemsFilteredType[][],
+		chartType: string
+	) => {
+		const isChecked = checkedArray.map(
 			(checked: boolean, index: number) => {
-				const productName = itemProducts[index]?.[0]?.item;
+				const filteredArray = items[index]?.[0]?.item;
 
-				if (currentFilterName === productName) {
+				if (currentFilterName === filteredArray) {
 					return !checked;
 				}
 
 				return checked;
 			}
 		);
 
-		return setCheckedStateProduct(isChecked);
+		const isRemainingStatuses =
+			currentFilterName !== claimsChartStatuses.settled;
+
+		const checkRemainingStatus = checkedArray.fill(
+			true,
+			0,
+			checkedArray.length - 1
+		);
```

Comment:
I need to understand why do you using this. Could you explain me?

Reply:
As "Remaining" is not a real status of the Claims Status picklist, I'm using this to set true to all the statuses but "settled" in the checkbox. "Settled" is the last status of the picklist.

Reply:
Thanks